agent/consul/autopilot/structs.go

```diff
@@ -20,6 +20,10 @@ type Config struct {
 	// be behind before being considered unhealthy.
 	MaxTrailingLogs uint64
 
+	// MinQuorum sets the minimum number of servers required in a cluster
+	// before autopilot can prune dead servers.
+	MinQuorum uint
```

Comment:
I think these should be ints instead, like with BootstrapExpect.

To avoid issues with someone providing a negative value you can do something like this:
https://github.com/hashicorp/consul/blob/e653583f698f4762f28535b251d0ec11910e65a8/agent/config/builder.go#L985

The issue with the uint is that is someone provides something like -1, then the config build fails with this error:
`cannot parse 'autopilot.min_quorum', -1 overflows uint`

That may seem cryptic to users, compared to how we validate other numbers that should always be positive.

Reply:
Discussed privately.